### DISCUSSION TITLE
Compiler Error CS0182 while setting ErrorMessage on MinLength Attribute

### DISCUSSION THREAD

**Discussion by robertodalmonte** [CommunityToolkit/dotnet#197] [Q&A] [answered]
In order to provide localized ErrorMessage to the user, I'm trying to use a localized string taken from a resx file.
I get an error (Compiler Error CS0182) that prevents the program to build (ErrorMessage = Resources.givenName).

```csharp
public partial class MainWindowViewModel : ObservableValidator
{
    [ObservableProperty] [Required] [MinLength(2, ErrorMessage = Resources.givenName)] [MaxLength(40)]
    private string _firstName = string.Empty;

    [ObservableProperty] [Required] [MinLength(2)] [MaxLength(40)]
    private string _lastName = string.Empty;
}
```

I guess I'm doing something wrong...is there a way to get the result I'm looking for?
Regards
Roberto

**Comment by Sergio0694** [comment, marked as answer]
This is unrelated from the MVVM Toolkit, the snippet above is just invalid C# syntax. As the docs for the error also state ([here](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0182)), as well as the ECMA spec for the C# language, only compile time constants are allowed as input for attributes. In your case, that `Resources.givenName` is not a constant, hence the error. You'll have to either make that a constant or come up with some other approach to wire this up (eg. a custom validation attribute taking the property name and then loading with reflection at runtime, or something) 🙂

**Reply by robertodalmonte** [reply]
Thank you, Sergio.
My question should have been: does the Community Toolkit offer a way to localize the ErrorMessage?
If I read you correctly, it does not in the way I hoped, but via a Custom Validation Attribute.
Is that correct?
BTW the community toolkit is supercalifragilisticexpialidocious
